.github/workflows/build.yaml

```diff
@@ -234,22 +260,54 @@ jobs:
           path: /mnt/cache/cc
           key: haos-cc-${{ matrix.board.id }}-${{ github.run_id }}
 
-      - name: Upload ova image to artifacts for test job
+      - name: Upload ova image to artifacts for the test job
         uses: actions/upload-artifact@v3
         if: ${{ matrix.board.id == 'ova' }}
         with:
           name: ova-image
           path: |
             output/images/haos_ova*.qcow2.xz
 
+      - name: Upload OS image artifact
+        uses: actions/upload-artifact@v3
+        if: ${{ github.event_name != 'release' && needs.prepare.outputs.publish_build != 'true' && matrix.board.id != 'ova' }}
+        with:
+          name: haos_${{ matrix.board.id }}-${{ needs.prepare.outputs.version_full }}.img.xz
+          path: |
+            output/images/haos_${{ matrix.board.id }}-${{ needs.prepare.outputs.version_full }}.img.xz
+
+      - name: Upload OS KVM image artifact
+        uses: actions/upload-artifact@v3
+        if: ${{ github.event_name != 'release' && needs.prepare.outputs.publish_build != 'true' && (matrix.board.id == 'generic-aarch64' || matrix.board.id == 'ova') }}
+        with:
+          name: haos_${{ matrix.board.id }}-${{ needs.prepare.outputs.version_full }}.qcow2.xz
+          path: |
+            output/images/haos_${{ matrix.board.id }}-${{ needs.prepare.outputs.version_full }}.qcow2.xz
+
+      - name: Upload OS VirtualBox/VMware image artifact
+        uses: actions/upload-artifact@v3
+        if: ${{ github.event_name != 'release' && needs.prepare.outputs.publish_build != 'true' && (matrix.board.id == 'generic-aarch64' || matrix.board.id == 'ova') }}
+        with:
+          name: haos_${{ matrix.board.id }}-${{ needs.prepare.outputs.version_full }}.vmdk.zip
+          path: |
+            output/images/haos_${{ matrix.board.id }}-${{ needs.prepare.outputs.version_full }}.vmdk.zip
+
+      - name: Upload RAUC update artifact
+        uses: actions/upload-artifact@v3
+        if: ${{ github.event_name != 'release' && needs.prepare.outputs.publish_build != 'true' }}
+        with:
+          name: haos_${{ matrix.board.id }}-${{ needs.prepare.outputs.version_full }}.raucb
+          path: |
+            output/images/haos_${{ matrix.board.id }}-${{ needs.prepare.outputs.version_full }}.raucb
+
```

Comment:
I guess there is no way using globbing or someting?

There are more images, a similar per image upload has been removed in 9f5b9f8afead535ac6c714718b2da862702d16ea.

Reply:
You can use glob or multiple paths - but in that case it creates a .zip archive of all the matched files. And that's yet another hidden _drawback_  (?) of using artifacts, it'll be a .zip archive every time. But at least GH sends the correct content-disposition headers, so you know it's a zip when you download it.

Reply:
Yeah zip is not ideal, as it gets big and you typically just want to download one.

But I think adding `vdi.zip` `vhdx.zip` and `ova` for completeness make sense still, or any particular reason those are not part of the upload?